### PR TITLE
doc/rados/troubleshooting: clarity and modernization

### PR DESCRIPTION
**Pull request by anthonyeleven**
doc/rados/troubleshooting: clarity and modernization

Various rewording for clarity and brevity.
Spelling and capitalization fixes.
Various updates to align with recent Ceph releases.

Signed-off-by: Anthony D'Atri <anthony.datri@gmail.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by anthonyeleven** [comment]
jenkins test make check

**Comment by anthonyeleven** [comment]
jenkins test api

**Comment by anthonyeleven** [comment]
@zdover23 This one's ready for you

File: doc/rados/troubleshooting/troubleshooting-mon.rst [hunk 1]
```diff
@@ -218,43 +222,45 @@ What if the state is ``probing``?
...
-  However, if you notice that the monitor jumps from ``synchronizing`` to
+  If you notice that the monitor jumps from ``synchronizing`` to
   ``electing`` and then back to ``synchronizing``, then you do have a
-  problem: the cluster state is advancing (i.e., generating new maps) way
-  too fast for the synchronization process to keep up. This used to be a
-  thing in early Cuttlefish, but since then the synchronization process was
-  quite refactored and enhanced to avoid just this sort of behavior. If this
-  happens in later versions let us know. And bring some logs
+  problem: the cluster state may be advancing (i.e., generating new maps)
+  too fast for the synchronization process to keep up. This was more common
```
Comment (zdover23) [inline doc/rados/troubleshooting/troubleshooting-mon.rst:246]: s/This was more common/This was a more common/

This phrase modifies the coming noun "thing", so we need the "a" in here.

File: doc/rados/troubleshooting/troubleshooting-mon.rst [hunk 2]
```diff
@@ -267,19 +273,20 @@ monitors::
       2: 127.0.0.1:6795/0 mon.c
       
 This may not be what you have however. For instance, in some versions of
-early Cuttlefish there was this one bug that could cause your ``monmap``
+early Cuttlefish there was a bug that could cause your ``monmap``
 to be nullified.  Completely filled with zeros. This means that not even
-``monmaptool`` would be able to read it because it would find it hard to
-make sense of only-zeros. Some other times, you may end up with a monitor
-with a severely outdated monmap, thus being unable to find the remaining
+``monmaptool`` would be able to make sense of cold, hard, inscrutable zeros.
```
Comment (zdover23) [inline doc/rados/troubleshooting/troubleshooting-mon.rst:278]: fancy, fancy. Let's leave this in. It has dash.

File: doc/rados/troubleshooting/troubleshooting-mon.rst [hunk 3]
```diff
@@ -321,38 +328,58 @@ Inject a monmap into the monitor
...
 
 What's the maximum tolerated clock skew?
 
-  By default the monitors will allow clocks to drift up to ``0.05 seconds``.
+  By default the monitors will allow clocks to drift up to 0.05 seconds (50 ms).
 
 
 Can I increase the maximum tolerated clock skew?
 
   This value is configurable via the ``mon-clock-drift-allowed`` option, and
```
Comment (zdover23) [inline doc/rados/troubleshooting/troubleshooting-mon.rst:361]: I think "This value" should be changed to "Maximum tolerated clock skew".

File: doc/rados/troubleshooting/troubleshooting-mon.rst [hunk 4]
```diff
@@ -321,38 +328,58 @@ Inject a monmap into the monitor
...
-  although you *CAN* it doesn't mean you *SHOULD*. The clock skew mechanism
-  is in place because clock skewed monitor may not properly behave. We, as
+  although you *CAN* you almost certainly *SHOULDN'T*. The clock skew mechanism
+  is in place because clock skewed monitors are liely to misbehave. We, as
```
Comment (zdover23) [inline doc/rados/troubleshooting/troubleshooting-mon.rst:363]: s/clock skewed/clock-skewed/

This is about as nit-picky as I get, but hyphenated phrasal participial-type things like this read much better when they have hyphens.

File: doc/rados/troubleshooting/troubleshooting-mon.rst [hunk 5]
```diff
@@ -321,38 +328,58 @@ Inject a monmap into the monitor
...
   developers and QA aficionados, are comfortable with the current default
   value, as it will alert the user before the monitors get out hand. Changing
-  this value without testing it first may cause unforeseen effects on the
-  stability of the monitors and overall cluster healthiness, although there is
-  no risk of dataloss.
-
+  this value may cause unforeseen effects on the
+  stability of the monitors and overall cluster health.
```
Comment (zdover23) [inline doc/rados/troubleshooting/troubleshooting-mon.rst:367]: This is great. This is a great edit.

I apologize if a comment like this feels like a waste of time, but this is a superb edit and I want to encourage this kind of edit.

File: doc/rados/troubleshooting/troubleshooting-mon.rst [hunk 6]
```diff
@@ -321,38 +328,58 @@ Inject a monmap into the monitor
...
-  detail`` should show something in the form of::
+  The monitors will warn you via the cluster status ``HEALTH_WARN``. ``ceph health
+  detail`` or ``ceph status`` should show something like::
 
       mon.c addr 10.10.0.1:6789/0 clock skew 0.08235s > max 0.05s (latency 0.0045s)
 
   That means that ``mon.c`` has been flagged as suffering from a clock skew.
 
+  On releases beginning with Luminous you can issue the
+  ``ceph time-sync-status`` command to check status.  Note that the loead mon
```
Comment (zdover23) [inline doc/rados/troubleshooting/troubleshooting-mon.rst:379]: s/loead/lead/

This is just a guess. Is "loaed" right?

Comment (zdover23) [reply doc/rados/troubleshooting/troubleshooting-mon.rst:379]: Nah, it's "lead".

File: doc/rados/troubleshooting/troubleshooting-osd.rst [hunk 7]
```diff
@@ -150,88 +162,134 @@ If you start your cluster and an OSD won't start, check the following:
...
+  Check ``syslog`` history for table fillage events.  You can mitigate this
+  bother by raising ``nf_conntrack_max`` to a much higher value via ``sysctl``.
+  Be sure to raise ``nf_conntrack_buckets`` accordingly to
+  ``nf_conntrack_max / 4``, which may require action outside of ``sysctl`` e.g.
+  ``"echo 131072 > /sys/module/nf_conntrack/parameters/hashsize``
+  More interdictive but fussier is to blacklist the associated kernel modules
+  to disable processing altogether.  This is fragile in that the modules
+  vary among kernel versions, as does the order in which they must be listed.
+  Even when blacklisted there are situations in which ``iptables`` or ``docker``
+  may active connection tracking anyway, so a "set and forget" strategy for
```
Comment (zdover23) [inline doc/rados/troubleshooting/troubleshooting-osd.rst:184]: s/active/activate/

File: doc/rados/troubleshooting/troubleshooting-osd.rst [hunk 8]
```diff
@@ -150,88 +162,134 @@ If you start your cluster and an OSD won't start, check the following:
...
 
 
 An OSD Failed
 -------------
 
-When a ``ceph-osd`` process dies, the monitor will learn about the failure
-from surviving ``ceph-osd`` daemons and report it via the ``ceph health``
-command::
+When a ``ceph-osd`` process dies, surviving ``ceph-osd`` daemons will report
+to the mons that it appears down, which will in tur surface the new status
```
Comment (zdover23) [inline doc/rados/troubleshooting/troubleshooting-osd.rst:210]: s/tur/turn/

File: doc/rados/troubleshooting/troubleshooting-osd.rst [hunk 9]
```diff
@@ -150,88 +162,134 @@ If you start your cluster and an OSD won't start, check the following:
...
+If the daemon stopped because of a heartbeat failure or ``suicide timeout``,
+the underlying drive or filesystem may be unresponsive. Check ``dmesg``
+and `syslog`  output for drive or other kernel errors.  You may need to
+specify something like ``dmesg -T`` to get timestamps, otherwise it's
+easy to mistake old errors for new.
 
 If the problem is a software error (failed assertion or other
-unexpected error), it should be reported to the `ceph-devel`_ email list.
+unexpected error), search the archives and tracker as above, and
+reported to the `ceph-devel`_ email list if there's no clear fix or
```
Comment (zdover23) [inline doc/rados/troubleshooting/troubleshooting-osd.rst:241]: s/reported/report it/

File: doc/rados/troubleshooting/troubleshooting-osd.rst [hunk 10]
```diff
@@ -245,16 +303,17 @@ Or::
...
-some placement group directories in the full OSD.
+If you cannot start a legacy Filestore OSD because it is full, you may reclaim
+some space deleting a few placement group directories in the full OSD.
 
 .. important:: If you choose to delete a placement group directory on a full OSD,
    **DO NOT** delete the same placement group directory on another full OSD, or
-   **YOU MAY LOSE DATA**. You **MUST** maintain at least one copy of your data on
-   at least one OSD.
+   **YOU WILL LOSE DATA**. You **MUST** maintain at least one copy of your data on
+   at least one OSD.  This is a rare and extreme intervention not to be undertaken
```
Comment (zdover23) [inline doc/rados/troubleshooting/troubleshooting-osd.rst:315]: s/rare and extreme intervention not/rare and extreme intervention, and is not/

I think this makes the sentence a little clearer.

File: doc/rados/troubleshooting/troubleshooting-osd.rst [hunk 11]
```diff
@@ -295,15 +354,17 @@ Check network statistics. ::
 Drive Configuration
 -------------------
 
-A storage drive should only support one OSD. Sequential read and sequential
-write throughput can bottleneck if other processes share the drive, including
-journals, operating systems, monitors, other OSDs and non-Ceph processes.
+A SAS or SATA storage drive should only house one OSD; NVMe drives readily
+handle two or more. Rread and write throughput can bottleneck if other processes
```
Comment (zdover23) [inline doc/rados/troubleshooting/troubleshooting-osd.rst:358]: s/Rread/Read/

File: doc/rados/troubleshooting/troubleshooting-osd.rst [hunk 12]
```diff
@@ -393,21 +458,23 @@ For more information, see `Filesystem Recommendations`_.
 Insufficient RAM
 ----------------
 
-We recommend 1GB of RAM per OSD daemon. You may notice that during normal
-operations, the OSD only uses a fraction of that amount (e.g., 100-200MB).
-Unused RAM makes it tempting to use the excess RAM for co-resident applications,
-VMs and so forth. However, when OSDs go into recovery mode, their memory
-utilization spikes. If there is no RAM available, the OSD performance will slow
-considerably.
+We recommend a *minimum* of 4GB of RAM per OSD daemon and sugget rounding up
```
Comment (zdover23) [inline doc/rados/troubleshooting/troubleshooting-osd.rst:461]: s/sugget/suggest/

File: doc/rados/troubleshooting/troubleshooting-osd.rst [hunk 13]
```diff
@@ -393,21 +458,23 @@ For more information, see `Filesystem Recommendations`_.
...
+from 6-8GB.  You may notice that during normal operations, ``ceph-osd``
+processes only uses a fraction of that amount (e.g., 100-200MB).
```
Comment (zdover23) [inline doc/rados/troubleshooting/troubleshooting-osd.rst:463]: s/uses/use/

File: doc/rados/troubleshooting/troubleshooting-osd.rst [hunk 14]
```diff
@@ -486,26 +554,47 @@ the internal code (such as passing data across locks into new threads).
...
+We have tradtionally recommended separate *public* (front-end) and *private*
+(cluster / back-end / replication) networks:
+
+#. Segregation of heartbeat and replication / recovery traffic (private)
+   from client and OSD <-> mon traffic (public).  This helps keep one
+   from DoS-ing the other, which could in turn result in a cascading failure.
+
+#. Additional throughput for both public and private traffic.
+
+When common networking technloogies were 100Mb/s and 1Gb/s this separation
```
Comment (zdover23) [inline doc/rados/troubleshooting/troubleshooting-osd.rst:569]: s|1Gb/s|1GB/s,|

(put a comma after "1GB/s")

File: doc/rados/troubleshooting/troubleshooting-osd.rst [hunk 15]
```diff
@@ -486,26 +554,47 @@ the internal code (such as passing data across locks into new threads).
...
+for just a public network:  with bonding (LACP) or equal-cost routing (e.g. FRR)
+you reap the benefits of increased throughput headroom, fault tolerance, and
+reduced OSD flapping.
+
+When a private network (or even a single host link) fails or degrades while the
+public network operates normally, OSDs may not handle this situation well. What
+happens is that OSDs use the public network to report each other ``down`` to
+the monitors, while marking themselves ``up``. The monitors then send out,
+again on the public network, an updated cluster map with affected OSDs marked
+`down`. These OSDs reply to the monitors "I'm not dead yet!", and the cycle
```
Comment (zdover23) [inline doc/rados/troubleshooting/troubleshooting-osd.rst:585]: This hearkens back to a time when every conversation was laced with Holy Grail references. Happily, it is again a very welcome thing.

**Review by zdover23 — COMMENTED** [review]
See the in-line comments in the "files changed" tab.

This is a massive undertaking, and an impressive addition.

File: doc/rados/troubleshooting/troubleshooting-osd.rst (hunk 14 above)
Comment (tchaikov) [reply doc/rados/troubleshooting/troubleshooting-osd.rst:569]: i think "100Mb/s and 1Gb/s" is correct. when it comes to bandwidth, we tend to use bits as the unit. they are also short for "100Mbs and 1Gbps".

File: doc/rados/troubleshooting/troubleshooting-mon.rst [hunk 16]
```diff
@@ -53,11 +58,9 @@ Initial Troubleshooting
   perform this for each monitor in the cluster. In section `Understanding
   mon_status`_ we will explain how to interpret the output of this command.
 
-  For the rest of you who don't tread on the bleeding edge, you will need to
-  ssh into the server and use the monitor's admin socket. Please jump to
-  `Using the monitor's admin socket`_.
-
-For other specific issues, keep on reading.
+  You may instead SSH into each mon node  and query daemon's admin socket.
```
[the remark this answers is not in the document]
Comment (anthonyeleven) [reply doc/rados/troubleshooting/troubleshooting-mon.rst:61]: Agreed, comrade.  I fumbled with the phrasing here a bit.  Actually the following two lines are superfluous given that the referenced section is immediately below, so I'm removing them.

File: doc/rados/troubleshooting/troubleshooting-mon.rst [hunk 17]
```diff
@@ -66,15 +69,16 @@ Using the monitor's admin socket
 The admin socket allows you to interact with a given daemon directly using a
 Unix socket file. This file can be found in your monitor's ``run`` directory.
 By default, the admin socket will be kept in ``/var/run/ceph/ceph-mon.ID.asok``
-but this can vary if you defined it otherwise. If you don't find it there,
-please check your ``ceph.conf`` for an alternative path or run::
+but this may be elsewhere if you have overridden the default directory. If you
+don't find it there, please check your ``ceph.conf`` for an alternative path or
```
[the remark this answers is not in the document]
Comment (anthonyeleven) [reply doc/rados/troubleshooting/troubleshooting-mon.rst:73]: Ack.

File: doc/rados/troubleshooting/troubleshooting-mon.rst [hunk 18]
```diff
@@ -66,15 +69,16 @@ Using the monitor's admin socket
...
+run::
 
   ceph-conf --name mon.ID --show-config-value admin_socket
 
-Please bear in mind that the admin socket will only be available while the
-monitor is running. When the monitor is properly shutdown, the admin socket
+Bear in mind that the admin socket will only be available while the monitor
```
[the remark this answers is not in the document]
Comment (anthonyeleven) [reply doc/rados/troubleshooting/troubleshooting-mon.rst:78]: `boldly go`.  Changed.

File: doc/rados/troubleshooting/troubleshooting-mon.rst [hunk 19]
```diff
@@ -66,15 +69,16 @@ Using the monitor's admin socket
...
+daemon is running. When the monitor is properly shutdown, the admin socket
```
[the remark this answers is not in the document]
Comment (anthonyeleven) [reply doc/rados/troubleshooting/troubleshooting-mon.rst:79]: Agreed.  I run into the same thing with "login" vs "log in".

File: doc/rados/troubleshooting/troubleshooting-mon.rst [hunk 20]
```diff
@@ -66,15 +69,16 @@ Using the monitor's admin socket
...
 will be removed. If however the monitor is not running and the admin socket
-still persists, it is likely that the monitor was improperly shutdown.
+persists, it is likely that the monitor was improperly shutdown.
```
[the remark this answers is not in the document]
Comment (anthonyeleven) [reply doc/rados/troubleshooting/troubleshooting-mon.rst:81]: ack

File: doc/rados/troubleshooting/troubleshooting-mon.rst [hunk 21]
```diff
@@ -170,10 +174,10 @@ How to troubleshoot this?
 
   First, make sure ``mon.a`` is running.
 
-  Second, make sure you are able to connect to ``mon.a``'s server from the
-  other monitors' servers. Check the ports as well. Check ``iptables`` on
-  all your monitor nodes and make sure you are not dropping/rejecting
-  connections.
+  Second, make sure you are able to connect to ``mon.a``'s node from the
+  other mon nodes. Check the TCP ports as well. Check ``iptables`` and
+  ``nf_conntrack`` on all nodes and make ensure that you are not
```
[the remark this answers is not in the document]
Comment (anthonyeleven) [reply doc/rados/troubleshooting/troubleshooting-mon.rst:179]: I did resist the urge to advise the reader to avail themselves of garlic fries ;)

File: doc/rados/troubleshooting/troubleshooting-mon.rst [hunk 22]
```diff
@@ -193,17 +197,17 @@ What if the state is ``probing``?
...
+  the monitor will pass through the probing state almost instantaneously.
+  In a multi-monitor cluster, the monitors will stay in this state until they
   find enough monitors to form a quorum -- this means that if you have 2 out
   of 3 monitors down, the one remaining monitor will stay in this state
   indefinitely until you bring one of the other monitors up.
 
-  If you have a quorum, however, the monitor should be able to find the
-  remaining monitors pretty fast, as long as they can be reached. If your
+  If you have a quorum the starting daemon should be able to find the
+  other monitors quickly, so long as they can be reached. If your
```
[the remark this answers is not in the document]
Comment (anthonyeleven) [reply doc/rados/troubleshooting/troubleshooting-mon.rst:210]: ack.  Adverbs are a lost art.

File: doc/rados/troubleshooting/troubleshooting-mon.rst [hunk 23]
```diff
@@ -218,43 +222,45 @@ What if the state is ``probing``?
...
 
-  This means the monitor is in the middle of an election. These should be
-  fast to complete, but at times the monitors can get stuck electing. This
-  is usually a sign of a clock skew among the monitor nodes; jump to
-  `Clock Skews`_ for more infos on that. If all your clocks are properly
-  synchronized, it is best if you prepare some logs and reach out to the
-  community. This is not a state that is likely to persist and aside from
+  This means the monitor is in the middle of an election. With recent Ceph
+  releases these typically complete quickly, but at times the monitors can
+  get stuck in what is known as an *election storm*. This can be
```
[the remark this answers is not in the document]
Comment (anthonyeleven) [reply doc/rados/troubleshooting/troubleshooting-mon.rst:227]: Ack.  Cleaned up a couple of following words at the same time.

File: doc/rados/troubleshooting/troubleshooting-mon.rst (hunk 1 above)
Comment (anthonyeleven) [reply doc/rados/troubleshooting/troubleshooting-mon.rst:246]: ack

File: doc/rados/troubleshooting/troubleshooting-mon.rst (hunk 2 above)
Comment (anthonyeleven) [reply doc/rados/troubleshooting/troubleshooting-mon.rst:278]: Hah.  I am at times influenced by this classic: http://www.catb.org/~esr/jargon/html/story-of-mel.html

File: doc/rados/troubleshooting/troubleshooting-mon.rst (hunk 3 above)
Comment (anthonyeleven) [reply doc/rados/troubleshooting/troubleshooting-mon.rst:361]: ack

File: doc/rados/troubleshooting/troubleshooting-mon.rst (hunk 4 above)
Comment (anthonyeleven) [reply doc/rados/troubleshooting/troubleshooting-mon.rst:363]: Ack.  There's history among these files of changing `filesystem` to `file system`; that one I disagree with and change back when I see it.

File: doc/rados/troubleshooting/troubleshooting-mon.rst (hunk 5 above)
Comment (anthonyeleven) [reply doc/rados/troubleshooting/troubleshooting-mon.rst:367]: ty.  I took an _Advanced Composition_ class in high school and was an editor of my college newspaper for a while.  Picked up great advice along the way, one of the rules I try to live by is that if a sentence doesn't lose meaning by removing a word, it shouldn't be there. 

Not a waste of time at all.

File: doc/rados/troubleshooting/troubleshooting-osd.rst (hunk 9 above)
Comment (anthonyeleven) [reply doc/rados/troubleshooting/troubleshooting-osd.rst:241]: Ack.  As I reworked this bit and others like it I wanted to encourage the reader to look for existing answers before going to the list or tracker.   Traffic on the list is already high; we don't need to make it worse.

File: doc/rados/troubleshooting/troubleshooting-osd.rst (hunk 10 above)
Comment (anthonyeleven) [reply doc/rados/troubleshooting/troubleshooting-osd.rst:315]: I think it's more dramatic as written, but don't object.

File: doc/rados/troubleshooting/troubleshooting-osd.rst (hunk 14 above)
Comment (anthonyeleven) [reply doc/rados/troubleshooting/troubleshooting-osd.rst:569]: I know the Google style guide favors commas like this.  I find it stilted but not worth arguing about.

File: doc/rados/troubleshooting/troubleshooting-osd.rst (hunk 15 above)
Comment (anthonyeleven) [reply doc/rados/troubleshooting/troubleshooting-osd.rst:585]: :D  The best part is that it's a perfect metaphor for this dynamic.

My book is laced with easter eggs, few of which alas people seem to pick up on.

File: doc/rados/troubleshooting/troubleshooting-osd.rst (hunk 14 above)
Comment (anthonyeleven) [reply doc/rados/troubleshooting/troubleshooting-osd.rst]: tchaikov: I think Zac meant only to suggest adding a comma.   I think `b` => `B` was inadvertent.

From my NSP days I instinctively wanted to write FE, GE, XE but I've found that those terms aren't widely recognized.

File: doc/rados/troubleshooting/troubleshooting-mon.rst (hunk 16 above)
[the remark this answers is not in the document]
Comment (zdover23) [reply doc/rados/troubleshooting/troubleshooting-mon.rst]: Sounds good.

File: doc/rados/troubleshooting/troubleshooting-mon.rst (hunk 23 above)
[the remark this answers is not in the document]
Comment (zdover23) [reply doc/rados/troubleshooting/troubleshooting-mon.rst]: Qapla!

File: doc/rados/troubleshooting/troubleshooting-mon.rst (hunk 2 above)
Comment (zdover23) [reply doc/rados/troubleshooting/troubleshooting-mon.rst:276]: I haven't thought of the story of Mel in a couple of years.

File: doc/rados/troubleshooting/troubleshooting-mon.rst (hunk 3 above)
Comment (zdover23) [reply doc/rados/troubleshooting/troubleshooting-mon.rst]: Also, "maximum tolerated clock skew" sounds awesome.

File: doc/rados/troubleshooting/troubleshooting-osd.rst (hunk 10 above)
Comment (zdover23) [reply doc/rados/troubleshooting/troubleshooting-osd.rst]: On repeated readings, I see your point.

**Review by zdover23 — APPROVED** [review]
LGTM. Proceed.

File: doc/rados/troubleshooting/troubleshooting-mon.rst (hunk 23 above)
[the remark this answers is not in the document]
Comment (anthonyeleven) [reply doc/rados/troubleshooting/troubleshooting-mon.rst]: A GitHub review in Klingon?  You risk being the subject of an XKCD cartoon ;)

**Pull request merged**